eox_tenant/monkey_patch/monkey_patch_proxys.py

```diff
@@ -0,0 +1,160 @@
+        proxy = True
+
+    def __unicode__(self):
+        key = getattr(settings, "EDNX_TENANT_KEY", "No tenant is active at the moment")
+        return u"<Tenant proxy as site_configuration: {}>".format(key)
+
+    @property
+    def enabled(self):
+        """
+        Return the value of USE_EOX_TENANT to activate site_configurations.
+        """
+        return getattr(settings, "USE_EOX_TENANT", False)
+
+    @enabled.setter
+    def enabled(self, value):
+        """
+        We ignore the setter since this is a read proxy.
+        """
+        pass
+
+    def get_value(self, name, default=None):
+        """
+        Return Configuration value from the Tenant loaded in the settings object
+        as if this was a SiteConfiguration class.
+        """
+        try:
+            return getattr(settings, name, default)
+        except AttributeError as error:
+            logger.exception("Invalid data at the TenantConfigProxy get_value. \n [%s]", error)
+
+        return default
+
+    def save(self, *args, **kwargs):
+        """
+        Don't allow to save TenantSiteConfigProxy model in database.
+        """
+        pass
+
+    @classmethod
+    def get_all_orgs(cls):
+        """
+        This returns a set of orgs that are considered within all microsites and TenantConfig.
+        This can be used, for example, to do filtering
+        """
+        # Check the cache first
+        org_filter_set = cache.get(TENANT_ALL_ORGS_CACHE_KEY)
+        if org_filter_set:
+            return org_filter_set
+
+        org_filter_set = set()
+        if not cls.has_configuration_set():
+            return org_filter_set
+
+        tenant_config = TenantConfig.objects.values_list("lms_configs")
+        microsite_config = Microsite.objects.values_list("values")  # pylint: disable=no-member
+
+        for config in chain(tenant_config, microsite_config):
+            try:
+                current = json.loads(config[0])
+                org_filter = current.get("course_org_filter", {})
+            except IndexError:
+                continue
+
+            if org_filter and isinstance(org_filter, list):
+                for org in org_filter:
+                    org_filter_set.add(org)
+            elif org_filter:
+                org_filter_set.add(org_filter)
+
+        cls.set_key_to_cache(TENANT_ALL_ORGS_CACHE_KEY, org_filter_set)
+
+        return org_filter_set
+
+    @classmethod
+    def get_value_for_org(cls, org, val_name, default=None):
+        """
+        Returns a configuration value for a microsite or TenantConfig which has an org_filter that matches
+        what is passed in.
+        """
+
+        if not cls.has_configuration_set():
+            return default
+
+        cache_key = "org-value-{}-{}".format(org, val_name)
+        cached_value = cache.get(cache_key)
+
+        if cached_value:
+            return cached_value
+
+        tenant_config = TenantConfig.objects.values_list("lms_configs")
```

Comment:
What is the final word on the caching of this queries? How many queries do we run for the Studio /home/ page when there are many courses of different orgs?